src/Npgsql/TypeHandlerRegistry.cs

```diff
@@ -115,7 +115,27 @@ internal static async Task Setup(NpgsqlConnector connector, NpgsqlTimeout timeou
             connector.TypeHandlerRegistry = new TypeHandlerRegistry(connector);
 
             if (!BackendTypeCache.TryGetValue(connector.ConnectionString, out var types))
-                types = BackendTypeCache[connector.ConnectionString] = await LoadBackendTypes(connector, timeout, async);
+            {
```

Comment:
I think that with today's Npgsql, turning on replication would automatically imply `ServerCompatibilityMode.NoTypeLoading` ([see the docs](http://www.npgsql.org/doc/connection-string-parameters.html#compatibility)). This means that no type loading query is issued, and only a fixed list of types are supported - this seems very appropriate for replication.